src/main/java/com/deque/axe/android/AxeView.java

```diff
@@ -383,6 +401,95 @@ public static boolean classNameIsOfType(
+                        className);
+        setCalculatedProps(
+                controlsCalculatedProps.calculatedNameProp(),
+                controlsCalculatedProps.calculatedRoleProp(),
+                controlsCalculatedProps.calculatedValueProp(),
+                controlsCalculatedProps.calculatedStateProp());
+        break;
+
+      case AndroidClassNames.EDIT_TEXT:
+        AxeCalculatedPropsImpl.EditTextCalculatedProps editTextCalculatedProps =
+                new AxeCalculatedPropsImpl.EditTextCalculatedProps(
+                        text,
+                        contentDescription,
+                        labelText,
+                        hintText,
+                        className);
+        setCalculatedProps(
+                editTextCalculatedProps.calculatedNameProp(),
+                editTextCalculatedProps.calculatedRoleProp(),
+                editTextCalculatedProps.calculatedValueProp(),
+                editTextCalculatedProps.calculatedValueProp());
+        break;
+
+      case AndroidClassNames.TEXT_VIEW:
+        AxeCalculatedPropsImpl.TextViewCalculatedProps textViewCalculatedProps =
+                new AxeCalculatedPropsImpl.TextViewCalculatedProps(
+                        text,
+                        contentDescription,
+                        className);
+
+        setCalculatedProps(
+                textViewCalculatedProps.calculatedNameProp(),
+                textViewCalculatedProps.calculatedRoleProp(),
+                textViewCalculatedProps.calculatedValueProp(),
+                textViewCalculatedProps.calculatedStateProp());
+        break;
+
+      case AndroidClassNames.IMAGE_VIEW:
+        AxeCalculatedPropsImpl.ImageViewCalculatedProps imageViewCalculatedProps =
+                new AxeCalculatedPropsImpl.ImageViewCalculatedProps(
+                        text,
+                        contentDescription,
+                        labelText,
+                        className);
+
+        setCalculatedProps(
+                imageViewCalculatedProps.calculatedNameProp(),
+                imageViewCalculatedProps.calculatedRoleProp(),
+                imageViewCalculatedProps.calculatedValueProp(),
+                imageViewCalculatedProps.calculatedStateProp());
+        break;
+
+      default:
+        AxeCalculatedPropsImpl.UnknownViewCalculatedProps unknownViewCalculatedProps =
+                new AxeCalculatedPropsImpl.UnknownViewCalculatedProps(
+                        text,
+                        contentDescription,
+                        className);
+
+        setCalculatedProps(
+                unknownViewCalculatedProps.calculatedNameProp(),
+                unknownViewCalculatedProps.calculatedRoleProp(),
+                unknownViewCalculatedProps.calculatedValueProp(),
+                unknownViewCalculatedProps.calculatedStateProp());
+        break;
+    }
+  }
+
+  private void setCalculatedProps(String name, String role, String value, String state) {
```

Comment:
Let's have a data class to wrap this map, that way you don't have to have the repeated code 
```
setCalculatedProps(
                unknownViewCalculatedProps.calculatedNameProp(),
                unknownViewCalculatedProps.calculatedRoleProp(),
                unknownViewCalculatedProps.calculatedValueProp(),
                unknownViewCalculatedProps.calculatedStateProp());
```
Every time you want to set the props on the object. Instead, you could call a constructor like:

```
AxeCalculatedPropsImpl.UnknownViewCalculatedProps unknownViewCalculatedProps =
                new AxeCalculatedPropsImpl.UnknownViewCalculatedProps(
                        text,
                        contentDescription,
                        className);

mCalculatedProps = CalculatedProps(unknownViewCalculatedProps)
```

I used the `m` there because this is a PR, please never write with m-prepended variables.